crispy_forms/layout.py

```diff
@@ -460,9 +460,9 @@ def __init__(self, *args, **kwargs):
 
     input_type = "reset"
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, name, value, *, css_id="", css_class=None, template=None, **kwargs):
```

Comment:
Oh my. I've spent _hours_ looking at this and getting muddled. The Fieldset class (next class down) has `css_id=None, css_class=""` which is the opposite of what this set of classes does.

I think we can make `css_id=None` and generate the `id` in the init method. Currently if a custom `id` is not provided the `id` gets generated on each render (including a slugify call). Let's do that in the `__init__` and avoid that. Separare PR for that.

I think ideally we want to make the signatures consistant. Having different `None` vs `""` is confusing for me, let alone others!

As it happens I guess it doesn't matter too much as they both get treated the same in templates.